tesseract-server/src/handlers/logic_layer/geoadjacents.rs

```diff
@@ -0,0 +1,351 @@
+use std::collections::HashMap;
+use std::str;
+
+use actix_web::{
+    HttpRequest,
+    HttpResponse,
+    Path,
+    Result as ActixResult,
+};
+use failure::{Error, format_err, bail};
+use lazy_static::lazy_static;
+use log::*;
+use serde_qs as qs;
+use serde_derive::Deserialize;
+use url::Url;
+
+use tesseract_core::names::{Cut, Property, LevelName, Mask};
+use tesseract_core::format::{format_records, FormatType};
+use tesseract_core::{MeaOrCalc, DataFrame, Column, ColumnData, is_same_columndata_type};
+use tesseract_core::schema::{Cube, DimensionType};
+use crate::app::AppState;
+use crate::errors::ServerError;
+use crate::logic_layer::{LogicLayerConfig, CubeCache, Time};
+use super::super::util;
+use crate::handlers::logic_layer::{query_geoservice, GeoserviceQuery};
+
+
+/// Handles default aggregation when a format is not specified.
+/// Default format is jsonrecords.
+pub fn logic_layer_default_geoadjacents_handler(
+    (req, _cube): (HttpRequest<AppState>, Path<()>)
+) -> ActixResult<HttpResponse>
+{
+    logic_layer_geoadjacent(req, "jsonrecords".to_owned())
+}
+
+
+/// Handles aggregation when a format is specified.
+pub fn logic_layer_geoadjacents_handler(
+    (req, cube_format): (HttpRequest<AppState>, Path<(String)>)
+) -> ActixResult<HttpResponse>
+{
+    logic_layer_geoadjacent(req, cube_format.to_owned())
+}
+
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct LogicLayerQueryOpt {
+    pub cube: String,
+    #[serde(flatten)]
+    pub cuts: Option<HashMap<String, String>>,
+    parents: Option<bool>,
+    debug: Option<bool>,
+    exclude_default_members: Option<bool>,
+    locale: Option<String>,
+    sparse: Option<bool>,
+}
```

Comment:
Feel free to remove `parents` and `sparse` since they're not being used. `locale` can also be removed as it does not make sense here. And I would remove `exclude_default_members` as well (we can add it back in case we need to support it).

Lastly, this query opt should have a more descriptive name.